File: java/resistor-color/src/main/java/ResistorColor.java
import java.util.Arrays;

class ResistorColor {

    private String[] colors = {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};

    int colorCode(String color) {
        return Arrays.asList(this.colors).indexOf(color);
    }

    String[] colors() {
        return this.colors.clone();
    }
}

File: c/resistor-color/resistor_color.c
#include <stdlib.h>
#include "resistor_color.h"

int color_code(resistor_band_t band) {

   int b = band;

   return b;
}

resistor_band_t* colors() {

   resistor_band_t * bands;
   bands = malloc(sizeof(resistor_band_t) * 10);
 
   for (resistor_band_t band = BLACK; band <= WHITE; band++) {
      bands[band] = band;
   }

   return bands;
}
